internal/response-consumer/handler_test.go

```diff
@@ -385,38 +469,268 @@ var _ = Describe("handler", func() {
-			Expect(run0.Status).To(Equal("running"))
-			hosts := fetchHosts(data[0].ID)
-			Expect(hosts).To(BeEmpty())
+				instance.onMessage(test.TestContext(), newSatResponseMessage(events, data.CorrelationID))
 
-			run1 := fetchRun(data[1].ID)
-			Expect(run1.Status).To(Equal("success"))
-			checkHost(data[1].ID, "success", nil, "")
+				events = buildSatEvents(
+					data.CorrelationID,
+					satPlaybookRunUpdateEvent(0, "localhost", "first console log\n"),
+					satPlaybookRunUpdateEvent(1, "localhost", "second console log"),
+					satPlaybookRunFinishedEvent("localhost", "success"),
+				)
+
+				instance.onMessage(test.TestContext(), newSatResponseMessage(events, data.CorrelationID))
+
+				run := fetchRun(data.ID)
+
+				Expect(run.Status).To(Equal("success"))
+				checkHost(data.ID, "success", utils.IntRef(1), "first console log\nsecond console log")
+			})
+		})
+
+		Describe("response_full false", func() {
+			It("concatenates the logs for the same host", func() {
+				var data = test.NewRun(accountNumber())
+				data.ResponseFull = false
+
+				Expect(db().Create(&data).Error).ToNot(HaveOccurred())
+
+				var hostData = test.NewRunHost(data.ID, "running", nil)
+				Expect(db().Create(&hostData).Error).ToNot(HaveOccurred())
+
+				events := buildSatEvents(
+					data.CorrelationID,
+					satPlaybookRunUpdateEvent(0, "localhost", "first console log\n"),
+				)
+
+				instance.onMessage(test.TestContext(), newSatResponseMessage(events, data.CorrelationID))
+
+				events = buildSatEvents(
+					data.CorrelationID,
+					satPlaybookRunUpdateEvent(1, "localhost", "second console log"),
+					satPlaybookRunFinishedEvent("localhost", "success"),
+				)
+
+				instance.onMessage(test.TestContext(), newSatResponseMessage(events, data.CorrelationID))
+
+				run := fetchRun(data.ID)
+
+				Expect(run.Status).To(Equal("success"))
+				checkHost(data.ID, "success", utils.IntRef(1), "first console log\nsecond console log")
+			})
+
+			It("adds indicator in logs for missed host sequence", func() {
+				var data = test.NewRun(accountNumber())
+				data.ResponseFull = false
+
+				Expect(db().Create(&data).Error).ToNot(HaveOccurred())
+
+				var hostData = test.NewRunHost(data.ID, "running", nil)
+				Expect(db().Create(&hostData).Error).ToNot(HaveOccurred())
+
+				events := buildSatEvents(
+					data.CorrelationID,
+					satPlaybookRunUpdateEvent(0, "localhost", "first console log\n"),
+				)
+
+				instance.onMessage(test.TestContext(), newSatResponseMessage(events, data.CorrelationID))
+
+				events = buildSatEvents(
+					data.CorrelationID,
+					satPlaybookRunUpdateEvent(6, "localhost", "second console log"),
+					satPlaybookRunFinishedEvent("localhost", "success"),
+				)
+
+				instance.onMessage(test.TestContext(), newSatResponseMessage(events, data.CorrelationID))
+
+				run := fetchRun(data.ID)
+
+				Expect(run.Status).To(Equal("success"))
+				checkHost(data.ID, "success", utils.IntRef(6), "first console log\n&#8230;second console log")
```

Comment:
@tahmidefaz remind me why we're using a HTML entity to encode the ellipsis? Also, it may be nicer to add it on a new line, e.g.
```
output line
…
another line
```

instead of 

```
output line
…another line
```

Reply:
It should be the unicode ellipsis character (`U+2026`). I will change it :+1:

Reply:
done!